src/decoder/wav.rs

```diff
@@ -44,7 +44,7 @@ where
         let channels = spec.channels;
 
         let total_duration = {
-            let sample_rate = sample_rate as u64;
```

Comment:
good catch!

It might not be best to have `sample_rate = sample_rate *channels` since that is not really the sample rate anymore. Maybe we can call it data_rate? 

Alternatively I kinda like  `len / sample_rate / channels` instead of `len / sample_rate` though the nano's line would get more complex then. We could rewrite that as: `nanos = (len - secs) * 1_000_000_000 / sample_rate / channels`.

I'm leaning to just renaming it to data_rate.

Reply:
> I'm leaning to just renaming it to data_rate.

Me too. It saves the hassle of preventing overflowing `nanos` again.